pkg/auth/tokenparser_test.go

```diff
@@ -321,4 +321,39 @@ func (s *TestTokenParserSuite) TestTokenParser() {
 		require.Equal(s.T(), email0, claims.Email)
 		require.Equal(s.T(), "OriginalSubValue:1234-ABCD", claims.OriginalSub)
 	})
+
+	s.Run("parse valid token with aud claim", func() {
+		username0 := uuid.Must(uuid.NewV4()).String()
+		identity0 := &authsupport.Identity{
+			ID:       uuid.Must(uuid.NewV4()),
+			Username: username0,
+		}
+		email0 := identity0.Username + "@email.tld"
+
+		tests := map[string]struct {
+			aud []string
+		}{
+			"single string": {
+				aud: []string{"aud-claim-1"},
```

Comment:
Not sure "if" and "how", but should we check instead that it supports single string "aud-claim-1" and not a slice containing a single element. I think this is what the rfc says: https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3 and this is what caused the issue in our case.

Reply:
The auth field is a string array in the RegisteredClaims: https://pkg.go.dev/github.com/golang-jwt/jwt/v5#ClaimStrings
So we can't set it to a single string on the golang RegisteredClaims struct level.
However when a token is parsed by the jwt v 5 lib it takes into account that the encoded claim can be a string and the parser converts it to a string array to put it into the RegisteredClaims.Audience field.

The only way for us to test it would be to generate a raw token and then parse it but then we would test the jwt library. Not our code. And we don't want to start testing our dependencies. We just need to make sure that we are using the right dependency and this test covers this.

Reply:
maybe an e2e test case that set's the `aud` field to a string and one that set's it to a list could help testing this use case. Not sure how much value that can bring.

Reply:
The jwt v5 will always generate a token with an array. We can't use it for generating a token with a string with this. We would have to generate and sign a token manually and then hardcode it in the test. Or use the old jwt lib for that. But again that would test the jwt lib itself. I don't think we should go this route of start testing 3rd party libs. Especially when it's quite tricky.